gpkit/nomials/nomial_core.py

```diff
@@ -0,0 +1,137 @@
+                elif x < 0:
+                    neg_vars.append((var.latex(excluded), x))
+
+            pvarstrs = ['%s^{%.2g}' % (varl, x) if "%.2g" % x != "1" else varl
+                        for (varl, x) in pos_vars]
+            nvarstrs = ['%s^{%.2g}' % (varl, -x)
+                        if "%.2g" % -x != "1" else varl
+                        for (varl, x) in neg_vars]
+            pvarstrs.sort()
+            nvarstrs.sort()
+            pvarstr = ' '.join(pvarstrs)
+            nvarstr = ' '.join(nvarstrs)
+            c = mag(c)
+            cstr = "%.2g" % c
+            if pos_vars and (cstr == "1" or cstr == "-1"):
+                cstr = cstr[:-1]
+            else:
+                cstr = latex_num(c)
+
+            if not pos_vars and not neg_vars:
+                mstrs.append("%s" % cstr)
+            elif pos_vars and not neg_vars:
+                mstrs.append("%s%s" % (cstr, pvarstr))
+            elif neg_vars and not pos_vars:
+                mstrs.append("\\frac{%s}{%s}" % (cstr, nvarstr))
+            elif pos_vars and neg_vars:
+                mstrs.append("%s\\frac{%s}{%s}" % (cstr, pvarstr, nvarstr))
+
+        if "units" in excluded:
+            return " + ".join(sorted(mstrs))
+
+        units = unitstr(self.units, r"\mathrm{~\left[ %s \right]}", "L~")
+        units_tf = units.replace("frac", "tfrac").replace(r"\cdot", r"\cdot ")
+        return " + ".join(sorted(mstrs)) + units_tf
+
+    __hash__ = NomialData.__hash__  # required by Python 3
+
+    @property
+    def value(self):
+        """Self, with values substituted for variables that have values
+
+        Returns
+        -------
+        float, if no symbolic variables remain after substitution
+        (Monomial, Posynomial, or Nomial), otherwise.
+        """
+        p = self.sub(self.values)
+        if len(p.exps) == 1:
+            if not p.exp:
+                return p.c
+        return p
+
+    def prod(self):
+        return self
+
+    def sum(self):
+        return self
+
+    def __ne__(self, other):
+        return not self.__eq__(other)
+
+    def __eq__(self, other):
+        """Equality test
+
+        Returns
+        -------
+        bool
+        """
+        if isinstance(other, Numbers):
+            return (len(self.exps) == 1 and  # single term
+                    not self.exps[0] and     # constant
+                    self.cs[0] == other)     # the right constant
+        return super(Nomial, self).__eq__(other)
+
+    def __float__(self):
+        if len(self.exps) == 1:
+            if not self.exps[0]:
+                return mag(self.c)
+        else:
+            raise AttributeError("float() can only be called on"
+                                 " monomials with no variable terms")
+
+    def __rmul__(self, other):
+        return self * other
+
+    def __truediv__(self, other):
+        """Support the / operator in Python 3.x"""
+        return self.__div__(other)
```

Comment:
that's a strange inheritance structure -- the child implements a method that the parent calls?